custom-options-plus.php

```diff
@@ -112,6 +113,77 @@ function cop_insert( $row ) {
 	);
 }
 
+
+function cop_get_last_available_id(){
+	global $wpdb, $COP_TABLE;
+	$dbname = $wpdb->dbname;
+	$result = $wpdb->get_results("
+		SELECT `AUTO_INCREMENT`
+		FROM  INFORMATION_SCHEMA.TABLES
+		WHERE TABLE_SCHEMA = '$dbname'
+		AND TABLE_NAME = '$COP_TABLE';
+	");
+	return $result[0]->AUTO_INCREMENT;
+}
+function cop_check_id_exists($item){
+	global $wpdb, $COP_TABLE;
+	$result = $wpdb->get_results("
+		SELECT id from $COP_TABLE WHERE id = $item->id AND label != '$item->label';
+	");
+	return count($result) ? true : false;
+}
+function cop_redefine_table_to_sequential(){
+	global $wpdb, $COP_TABLE;
+	$data = $wpdb->get_results("SELECT id, label FROM $COP_TABLE ORDER BY id ASC");
+	$id = 1;
+	$result_reset_ai = $wpdb->get_results("ALTER TABLE $COP_TABLE AUTO_INCREMENT = 1");
+	foreach($data as $option){
+		if( !($option->id == $id) ){
+			$wpdb->update(
+				$COP_TABLE,
+				array('id' => $id),
+				array('id' => $option->id)
+			);
+		}
+		$id++;
+	}
+}
+function cop_update_id($item){
+	global $wpdb, $COP_TABLE;
+	$wpdb->query("ALTER TABLE $COP_TABLE AUTO_INCREMENT = 1");
```

Comment:
yes, the first thought was use cop_update originally, but update ids is more simpler.